client/my-sites/site-settings/cloudflare-analytics/index.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { flowRight, partialRight, pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wrapSettingsForm from '../wrap-settings-form';
+import { Card } from '@automattic/components';
+import { isDesktop } from '@automattic/viewport';
+import { getPlugins } from 'calypso/state/plugins/installed/selectors';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextValidation from 'calypso/components/forms/form-input-validation';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import getCurrentRouteParameterized from 'calypso/state/selectors/get-current-route-parameterized';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import SettingsSectionHeader from 'calypso/my-sites/site-settings/settings-section-header';
+import { localizeUrl } from 'calypso/lib/i18n-utils';
+import cloudflareIllustration from 'calypso/assets/images/illustrations/cloudflare-logo.svg';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const validateTrackingId = ( code ) => ! code || code.match( /^(UA-\d+-\d+)|(G-[A-Z0-9]+)$/i ); // @TODO: how are cloudflare tracking IDs formatted?
+
+export class CloudflareAnalyticsSettings extends Component {
```

Comment:
Is there a reason why this needs to be a class component instead of a function component? That way we can use hooks. Although, I'm not sure if the codebase has preference over one or the other. I know some are opinionated to not use hooks.

Reply:
We can definitely use function components and hooks as much as we like in Calypso going forward.

Reply:
A lot of the code in this PR was lifted directly from the Google Analytics card directly below it which is the main reason, haha. It's a good idea to modernize it though, so I'll go ahead and do a bit of refactoring.